File: go/wam/8378.go
const WamGroupHistorySenderUserJourney = 8378 // channel: regular

type GroupHistorySenderUserJourneyEvent struct {
	BundleSendSource BUNDLESENDSOURCE `wam:"bundle_send_source"` // field 1
	GroupHistoryMessagesCount int64 `wam:"group_history_messages_count"` // field 2
	GroupHistoryOutWindowPinsCount int64 `wam:"group_history_out_window_pins_count"` // field 3
	GroupHistoryPinsCount int64 `wam:"group_history_pins_count"` // field 4
	GroupHistorySenderActionType GROUPHISTORYSENDERACTIONTYPE `wam:"group_history_sender_action_type"` // field 5
	GroupHistorySystemMessageType GROUPHISTORYSYSTEMMESSAGETYPE `wam:"group_history_system_message_type"` // field 6
	GroupHistoryUncountedMessagesCount int64 `wam:"group_history_uncounted_messages_count"` // field 7
	GroupSize int64 `wam:"group_size"` // field 8
	IneligibleReason GROUPHISTORYINELIGIBILITYREASON `wam:"ineligible_reason"` // field 9
	RecipientCount int64 `wam:"recipient_count"` // field 10
	UiSurface TSSURFACE `wam:"ui_surface"` // field 11
	UnifiedSessionId string `wam:"unified_session_id"` // field 12
	UserJourneyTs int64 `wam:"user_journey_ts"` // field 13
	IsTeeBotNoticeOnly bool `wam:"is_tee_bot_notice_only"` // field 14
	IsGroupHistoryToggledOn bool `wam:"is_group_history_toggled_on"` // field 15
	XIneligibleReasons string `wam:"x_ineligible_reasons"` // field 16
}

File: ts/wam/8378.ts
export const WamGroupHistorySenderUserJourney = 8378 as const

export interface GroupHistorySenderUserJourneyEvent {
  /** field 1, wire `bundle_send_source` */
  bundleSendSource?: BUNDLESENDSOURCE
  /** field 2, wire `group_history_messages_count` */
  groupHistoryMessagesCount?: number
  /** field 3, wire `group_history_out_window_pins_count` */
  groupHistoryOutWindowPinsCount?: number
  /** field 4, wire `group_history_pins_count` */
  groupHistoryPinsCount?: number
  /** field 5, wire `group_history_sender_action_type` */
  groupHistorySenderActionType?: GROUPHISTORYSENDERACTIONTYPE
  /** field 6, wire `group_history_system_message_type` */
  groupHistorySystemMessageType?: GROUPHISTORYSYSTEMMESSAGETYPE
  /** field 7, wire `group_history_uncounted_messages_count` */
  groupHistoryUncountedMessagesCount?: number
  /** field 8, wire `group_size` */
  groupSize?: number
  /** field 9, wire `ineligible_reason` */
  ineligibleReason?: GROUPHISTORYINELIGIBILITYREASON
  /** field 10, wire `recipient_count` */
  recipientCount?: number
  /** field 11, wire `ui_surface` */
  uiSurface?: TSSURFACE
  /** field 12, wire `unified_session_id` */
  unifiedSessionId?: string
  /** field 13, wire `user_journey_ts` */
  userJourneyTs?: number
  /** field 14, wire `is_tee_bot_notice_only` */
  isTeeBotNoticeOnly?: boolean
  /** field 15, wire `is_group_history_toggled_on` */
  isGroupHistoryToggledOn?: boolean
  /** field 16, wire `x_ineligible_reasons` */
  xIneligibleReasons?: string
}